packages/dds/tree/src/simple-tree/arrayNode.ts

```diff
@@ -837,7 +832,9 @@ abstract class CustomArrayNodeBase<const T extends ImplicitAllowedTypes>
 	): void {
 		const destinationField = getSequenceField(this);
 		if (destinationField.context === undefined) {
-			throw new UsageError(`An array cannot be mutated before being inserted into the tree`);
+			throw new UsageError(
+				`Cannot move elements of an array before the array is inserted into the tree`,
```

Comment:
Now it sounds like you're calling out the source array as being the problem. It very well could be (do you need a check for that in here?), but the destination seems to be problem here.
```suggestion
				`Cannot move elements into an array before the array is inserted into the tree`,
```

Reply:
Good call, updated on both accounts.